oadp-operator/pkg/credentials/credentials.go

```diff
@@ -19,43 +22,53 @@ const (
 
 var (
 	mountPropagationToHostContainer = corev1.MountPropagationHostToContainer
-	cloudProvidersPluginFields      = map[oadpv1alpha1.DefaultPlugin]CloudProviderFields{
+	pluginSpecificFields            = map[oadpv1alpha1.DefaultPlugin]DefaultPluginFields{
 		oadpv1alpha1.DefaultPluginAWS: {
 			secretName:         "cloud-credentials",
 			mountPath:          "/credentials",
 			envCredentialsFile: "AWS_SHARED_CREDENTIALS_FILE",
+			pluginImage:        fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_AWS_PLUGIN_REPO"), os.Getenv("VELERO_AWS_PLUGIN_TAG")),
 		},
 		oadpv1alpha1.DefaultPluginGCP: {
 			secretName:         "cloud-credentials-gcp",
 			mountPath:          "/credentials-gcp",
 			envCredentialsFile: "GOOGLE_APPLICATION_CREDENTIALS",
+			pluginImage:        fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_GCP_PLUGIN_REPO"), os.Getenv("VELERO_GCP_PLUGIN_TAG")),
 		},
 		oadpv1alpha1.DefaultPluginMicrosoftAzure: {
 			secretName:         "cloud-credentials-azure",
 			mountPath:          "/credentials-azure",
 			envCredentialsFile: "AZURE_CREDENTIALS_FILE",
+			pluginImage:        fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_AZURE_PLUGIN_REPO"), os.Getenv("VELERO_AZURE_PLUGIN_TAG")),
+		},
+		oadpv1alpha1.DefaultPluginOpenShift: {
+			pluginImage: fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_OPENSHIFT_PLUGIN_REPO"), os.Getenv("VELERO_OPENSHIFT_PLUGIN_TAG")),
+		},
+		oadpv1alpha1.DefaultPluginCSI: {
+			//TODO: Check if the Registry needs to an upstream one from CSI
+			pluginImage: fmt.Sprintf("%v/%v/%v:%v", os.Getenv("REGISTRY"), os.Getenv("PROJECT"), os.Getenv("VELERO_CSI_PLUGIN_REPO"), os.Getenv("VELERO_CSI_PLUGIN_TAG")),
 		},
 	}
 )
 
 func AppendCloudProviderVolumes(velero *oadpv1alpha1.Velero, ds *appsv1.DaemonSet) error {
 	var veleroContainer *corev1.Container
 	// Find Velero container
-	for _, container := range ds.Spec.Template.Spec.Containers {
+	for i, container := range ds.Spec.Template.Spec.Containers {
 		if container.Name == common.Velero {
-			veleroContainer = &container
+			veleroContainer = &ds.Spec.Template.Spec.Containers[i]
```

Comment:
How is this different?
`container` is equivalent to `ds.Spec.Template.Spec.Containers[i]`? Or is it just a copy?

Reply:
@kaovilai yes `container` is different than `ds.Spec.Template.Spec.Containers[i]`, `container` is just mutated representation of the actual object in a temp variable accessible inside the `for` loop.

Reply:
Yeah inside a loop that var is just a copy. So this change looks correct

Reply:
They're certainly equivalent in content. But since we need a pointer, we need the original. The `range` operator assigns the value of `Containers[i]` to a copy -- the local variable `container`, so `&container` returns the address of the local variable rather than the original:
https://medium.com/@betable/3-go-gotchas-590b8c014e0a